Add preferences dialogue and implement toggle for folder contents reordering

daPhipz opened the pull request:
This is a very noob approach at adding a settings dialogue.
I mostly took the code from [here](https://www.codeproject.com/Articles/5271677/How-to-Create-A-GNOME-Extension) and adapted it to this extension. There is a tutorial series on YouTube from the same guy who wrote the blog post [here](https://www.youtube.com/playlist?list=PLr3kuDAFECjZhW-p56BoVB7SubdUHBVQT). Parts 4, 6 and 7 are about this topic, maybe give them a watch if you can handle his rather slow speed of explaining things xD

As I am fairly new to developing extensions and coding in general, I am afraid that I don't know how to make the extension listen to the settings dialogue and adjusting its behaviour accordingly. I know that you can read and write settings with `gridReorder.settings.get_XXX()` and `gridReorder.settings.set_XXX()` (see blog post linked above), but that's about it xD

I hope that you still find this useful and see it as a basis to build things off of!
What do you think?
As usual, you can commit directly to my branch if necessary.

stuarthayhurst commented:
I was planning on implementing the preferences dialogue and a gsettings key for toggling folder contents reordering. I'll look through the code and add the folder contents setting. Thanks :)

stuarthayhurst commented:
That's a bit of clean up so the new additions fit better with the style for the rest of the project. I also added `prefs.ui` to the packed extension, and added the compiled gsettings schema to `make clean`. I've taken out the gsettings code about reordering the position of folders, as I don't really want UI options that do nothing for the meantime, although once this is finished, I can begin to implement that. Also, the prefs menu displays now, but it's not yet hooked up the the code.

daPhipz commented:
> I also added prefs.ui to the packed extension

Oh shoot - I deleted that bit in order to have nice, clean commits, completely forgot to add it back in! Thanks xD

stuarthayhurst commented:
Ok, so I've managed to get the settings menu to reflect the state of the extension's gsettings values. If I can set the gsettings values from the settings menu, we'll have a working settings menu for configuring the extension

stuarthayhurst commented:
Implementing the effect of the extension's settings should be pretty easy, it's no different to the gsettings code already in the extension

stuarthayhurst commented:
That's done, now the slider matches the gsettings value when the settings box loads, and when the slider is toggled, the gsettings value is also toggled.

stuarthayhurst commented:
I'm happy with the merge request, if you're happy as well I'll go ahead and merge it.
Thanks :)

daPhipz commented:
> and when the slider is toggled, the gsettings value is also toggled.

Can confirm for 3.38, I am happy with merging to `master`! :tada:
